src/skfolio/distribution/multivariate/_vine_copula.py

```diff
@@ -0,0 +1,1145 @@
+
+                    if min_value >= max_value:
+                        raise ValueError(
+                            "The conditioning tuple lower bound must be lower than "
+                            "its upper bound."
+                        )
+                    initial_cond[var] = (min_value, max_value)
+
+                    if self.log_transform:
+                        if not np.isneginf(min_value):
+                            min_value = np.log(1 + min_value)
+                        if not np.isposinf(max_value):
+                            max_value = np.log(1 + max_value)
+                    if self.fit_marginals:
+                        # Transform the bounds using the marginal CDF
+                        dist = self.marginal_distributions_[var]
+                        u_min = 0.0 if np.isneginf(min_value) else dist.cdf(min_value)
+                        u_max = 1.0 if np.isposinf(max_value) else dist.cdf(max_value)
+                    else:
+                        u_min, u_max = min_value, max_value
+
+                    # Sample uniformly in the transformed interval.
+                    samples = rng.uniform(low=u_min, high=u_max, size=n_samples)
+
+                elif np.isscalar(value):
+                    initial_cond[var] = value
+                    if self.log_transform:
+                        value = np.log(1 + value)
+                    if self.fit_marginals:
+                        # Transform conditioning value using the fitted marginal CDF.
+                        value = self.marginal_distributions_[var].cdf(value)
+                    samples = np.full(n_samples, value)
+                else:
+                    samples = np.asarray(value)
+                    initial_cond[var] = samples
+                    if samples.ndim != 1 or samples.shape[0] != n_samples:
+                        raise ValueError(
+                            "When an array is provided in `conditioning`, it must be a "
+                            f"1D array of length n_samples={n_samples}, got {samples.ndim}D of "
+                            f"length {samples.shape[0]}"
+                        )
+                    if self.log_transform:
+                        samples = np.log(1 + samples)
+                    if self.fit_marginals:
+                        # Transform conditioning samples using the fitted marginal CDF.
+                        samples = self.marginal_distributions_[var].cdf(samples)
+                uniform_cond_samples[var] = samples
+
+        # Determine sampling order based on vine structure.
+        sampling_order = self._sampling_order(conditioning_vars=conditioning_vars)
+
+        # Generate independent Uniform(0,1) samples for non-conditioned nodes.
+        X_rand = iter(rng.random(size=(n_samples, n_assets - len(conditioning_vars))).T)
+
+        # Initialize samples for each node according to the sampling order.
+        queue: deque[tuple[Node, bool]] = deque()
```

Comment:
I think L577-630 deserves to be put in a private method like `self._do_sample(X_rand, sampling_order)`. This would make the flow of `sample()` more obvious and encapsulate the core logic, like with `_sampling_order`. WDYT?